src/main/java/cubicchunks/worldgen/generator/custom/populator/PopulatorUtils.java

```diff
@@ -75,17 +75,18 @@ public static void genOreUniform(ICubicWorld world, CustomGeneratorSettings cfg,
         }
     }
 
-    public static void genOreGaussian(ICubicWorld world, CustomGeneratorSettings cfg, Random random, CubePos pos,
-            int count, double probability, WorldGenerator generator, double mean, double stdDev) {
+    public static void genOreGaussian(ICubicWorld world, CustomGeneratorSettings cfg, Random random, CubePos pos, int count,
+                                      double probability, WorldGenerator generator, int mean, double stdDev, int spacing, int heightLimit) {
         for (int i = 0; i < count; ++i) {
-            if (random.nextDouble() > probability) {
-                continue;
-            }
             int yOffset = random.nextInt(Cube.SIZE) + Cube.SIZE / 2;
             int blockY = pos.getMinBlockY() + yOffset;
-            //noinspection SuspiciousNameCombination
-            double prob = MathUtil.gaussianProbabilityDensity(blockY, mean, stdDev);
-            if (random.nextDouble() > prob) {
+            //skip all potential spawns above the height limit
+            if(blockY > heightLimit){
+                continue;
+            }
+            double modifier = MathUtil.gaussianProbabilityCyclic(blockY, mean, stdDev, spacing);
+            //Modify base probability with the curve
+            if (random.nextDouble() > (probability * modifier)) {
```

Comment:
How did you ensure that peak average amount of blocks/area is the same as in vanilla? This is obvious for  uniform ones. Not so obvious from this (and even my code got it wrong, but only because I forgot to scale it for height). And I know it may be a bit too much especially when I didn't do it myself, but this is one of the rare cases in world generation that can be unit tested.